azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -88,6 +88,23 @@ protected void doExecute() throws AzureExecutionException {
         updateTelemetryProperties();
     }
 
+    private void doValidate() throws AzureExecutionException {
+        validateConfiguration(message -> AzureMessager.getMessager().error(message.getMessage()), true);
+        validateArtifactCompileVersion();
```

Comment:
`validateArtifactCompileVersion()` should be part of `validateConfiguration`

Reply:
`validateConfiguration` is the schema-based validation to validate the configuration values, while validateArtifactCompileVersion is to validate user's artifact